src/shared/smbios11.c

```diff
@@ -33,11 +33,7 @@ int read_smbios11_field(unsigned i, size_t max_size, char **ret_data, size_t *re
 
         assert_cc(offsetof(struct dmi_field_header, contents) == 5);
 
-        r = read_virtual_file(
-                        p,
-                        max_size >= SIZE_MAX - offsetof(struct dmi_field_header, contents) ? SIZE_MAX :
-                        sizeof(dmi_field_header) + max_size,
```

Comment:
Hmm? Why is this dropped? Previously, the `max_size` param represents the max content size, now it might become a short read for caller if content size > max_size - sizeof(dmi_field_header)